packages/protocol/contracts/team/airdrop/MerkleClaimable.sol

```diff
@@ -39,6 +44,7 @@ abstract contract MerkleClaimable is OwnableUpgradeable {
         bytes32[] calldata proof
     )
         external
+        nonReentrant
```

Comment:
I think this is not necessary, we solved it 'in-house' already.
A contract is vulnerable if it `calls`, `sends` or `transfers` to an untrusted address (malicious contract with fallback function).
A design to prevent that is check-effects interaction. (https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html)
We already have that given (by line 52 + line 58), because the only place an untrusted address can be invoked, is within `_claimWIthData()` but we already marked the hash as `used`.

Reply:
If you are sending an ERC1155 token to an target address and the target address is a contract and implement a ERC1155 callback, then the callback can call this contract to re-enter.

Reply:
> If you are sending an ERC1155 token to an target address and the target address is a contract and implement a ERC1155 callback, then the callback can call this contract to re-enter.

Yes. But it would revert because the hash is already “used”. (Line 60)